internal/controller/etcdopstask/handler/ondemandsnapshot/ondemandsnapshot.go

```diff
@@ -0,0 +1,224 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ondemandsnapshot
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask/handler"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrGetEtcd represents the error in case of fetching etcd object. Reason could be internal error
+	ErrGetEtcd druidv1alpha1.ErrorCode = "ERR_GET_ETCD"
+	// ErrEtcdNotReady represents the error in case etcd is not ready
+	ErrEtcdNotReady druidv1alpha1.ErrorCode = "ERR_ETCD_NOT_READY"
+	// ErrBackupNotEnabled represents the error in case backup is not enabled
+	ErrBackupNotEnabled druidv1alpha1.ErrorCode = "ERR_BACKUP_NOT_ENABLED"
+	// ErrGetSecret represents the error in case of failure in fetching the secret
+	ErrGetSecret druidv1alpha1.ErrorCode = "ERR_GET_SECRET"
```

Comment:
Can we rename this to something like this: `ErrGetCASecret` etc. as currently it's feel like it's error code for backup secret failure which is not the case, wdyt ?